src/Sarif/Baseline/ResultMatching/DataStructures/MatchedResults.cs

```diff
@@ -161,16 +157,14 @@ public Result CalculateBasedlinedResult(DictionaryMergeBehavior propertyBagMerge
             Dictionary<string, object> firstPropertyBag,
             Dictionary<string, object> secondPropertyBag)
         {
-            Dictionary<string, object> result = firstPropertyBag;
+            Dictionary<string, object> mergedPropertyBag = firstPropertyBag;
 
-            foreach (string key in secondPropertyBag.Keys)
+            foreach (string key in secondPropertyBag.Keys.Except(firstPropertyBag.Keys))
```

Comment:
>Except [](start = 58, length = 6)

Avoids hand-coded loop.

Reply:
Is there a perf loss in the new implementation? It looks like Except might create another list and the keys are enumerated more than once. So if the lists are large, there could be a perf problem. But I'm just guessing and I don't know if perf is a consideration for this method.

---
In reply to: [296356649](https://github.com/microsoft/sarif-sdk/pull/1542#discussion_r296356649) [](ancestors = 296356649)

Reply:
Good question. I checked the source code. It creates a `Set` to perform the existence checking, but it only iterates once:

[https://github.com/dotnet/corefx/blob/master/src/System.Linq/src/System/Linq/Except.cs](https://github.com/dotnet/corefx/blob/master/src/System.Linq/src/System/Linq/Except.cs)



---
In reply to: [296895448](https://github.com/microsoft/sarif-sdk/pull/1542#discussion_r296895448) [](ancestors = 296895448,296356649)

Reply:
Why would we want the set to be created in this case? Is there something that you find objectionable about the original code? Can you articulate it?

---
In reply to: [296915286](https://github.com/microsoft/sarif-sdk/pull/1542#discussion_r296915286) [](ancestors = 296915286,296895448,296356649)